library/modules/Translation.cpp

```diff
@@ -167,6 +174,23 @@ string Translation::TranslateName(const df::language_name * name, bool inEnglish
     CHECK_NULL_POINTER(name);
 
     string out;
+
+    auto fp = df::global::translate_name;
+
+    if (fp) {
+        DEBUG(translate).print("using df provided translate_name function\n");
+        typedef std::function<void(const df::language_name&, std::string&, bool, bool)> fun_type;
+        auto f = reinterpret_cast<fun_type*>(fp);
```

Comment:
this would be terrible to require all callers to do. can we make codegen produce a more usable type signature?

Reply:
not easily

Reply:
That would be sufficient, yeah, and it would give us a place to wrap the functionality if we need to, like with cancel_job.